Documentation/contributor-guide/features.md

```diff
@@ -0,0 +1,39 @@
+# Features 
+
+This document provides guidelines for adding and removing etcd features. The etcd features fall into two stages, experimental and stable. 
+
+## Experimental
+
+Any new feature should be added as an experimental feature. An experimental feature is characterized as below:
+- Associated with an issue with a clear need for such a feature.
+- Any configuration flags related to the implementation of the feature are prefixed with `experimental` e.g. `--experimental-feature-name`.
+- Any variable names related to the implementation of the feature are prefixed with `Experimental` e.g. `ExperimentalFeatureName`.
+- Might be buggy. Enabling the feature may not work as expected. 
+- Support for such a feature may be dropped at any time without notice.
+- Disabled by default when added initially.
+
+### Graduation to Stable
+
+It is important that experimental features don't get stuck in that stage.  An experimental feature should move to the stable stage by following the lifecycle steps below:
+- A new feature added in the latest release and disabled by default e.g. etcd v3.5 or v4.0
+- Given there are no open issues against the feature,
+  - It is enabled by default in the next release e.g. etcd v3.6 or v4.1
+  - It move to stable stage in the following release e.g. etcd v3.7 or v4.2. If release cycle takes longer, one year time frame can be used to move to the stable stage.
```

Comment:
In general, I agree with the two points here. But I think two minor releases is just a minimum requirement. It means it isn't necessarily to graduate an experimental feature after two minor releases (e.g., added in 3.5, and graduate in 3.7), although usually it does. 

The point is we need to have some flexibility on this.

Reply:
@ahrtr I agree that flexibility is needed. How about re-write as below for clarity that the next action is based on the review?: 
- Given there are no open issues against the feature, it is ready for review to, 
  - Enable by default in the next release e.g. etcd v3.6 or v4.1
  - Move to the stable stage in the following release e.g. etcd v3.7 or v4.2. If release cycle takes longer, one year time frame can be used for the review.

Reply:
`it is ready for review` makes sense, since we have flexibility to approve or reject the PR.

Reply:
Yup, thanks @ahrtr